Content/GUI/Quests/QuestsUIState.cs

```diff
@@ -0,0 +1,84 @@
+﻿using PathOfTerraria.Content.GUI.Utilities;
+using Terraria.Localization;
+
+namespace PathOfTerraria.Content.GUI.Quests;
+
+internal class QuestsUIState : DraggableSmartUi
+{
+	private QuestsCompletedInnerPanel _questsCompletedInnerPanel;
+	private QuestsInProgressInnerPanel _questsInProgressInnerPanel;
+
+	public void Toggle()
+	{
+		if (IsVisible)
+		{
+			IsVisible = false;
+			return;
+		}
+
+		RemoveAllChildren();
+		CreateMainPanel();
+		AddPassiveTreeInnerPanel();
+		AddSkillsTreeInnerPanel();
+		AddCloseButton();
+
+		IsVisible = true;
+	}
+
+	public override void Draw(SpriteBatch spriteBatch)
+	{
+		Recalculate();
+		base.Draw(spriteBatch);
+	}
+
+	protected void CreateMainPanel()
+	{
+		var localizedTexts = new (string key, LocalizedText text)[]
+		{
+			("QuestAvailableTab", Language.GetText("Mods.PathOfTerraria.GUI.QuestAvailableTab")),
+			("QuestCompletedTab", Language.GetText("Mods.PathOfTerraria.GUI.QuestCompletedTab"))
+		};
+		Panel = new UIDraggablePanel(false, false, localizedTexts, DraggablePanelHeight);
+		Panel.OnActiveTabChanged += HandleActiveTabChanged;
+		Panel.Left.Set(LeftPadding, 0.5f);
+		Panel.Top.Set(TopPadding, 0.5f);
+		Panel.Width.Set(PanelWidth, 0);
+		Panel.Height.Set(PanelHeight, 0);
+		Append(Panel);
+	}
+
+	private void HandleActiveTabChanged()
+	{
+		switch (Panel.ActiveTab)
+		{
+			case "QuestAvailableTab":
+				_questsCompletedInnerPanel.Visible = false;
+				_questsInProgressInnerPanel.Visible = true;
+				break;
+			case "QuestCompletedTab":
+				_questsCompletedInnerPanel.Visible = true;
+				_questsInProgressInnerPanel.Visible = false;
+				break;
+		}
+	}
+
+	protected void AddPassiveTreeInnerPanel()
+	{
+		_questsCompletedInnerPanel = new QuestsCompletedInnerPanel();
+		_questsCompletedInnerPanel.Left.Set(0, 0);
```

Comment:
This is not an issue, but setting left to 0, 0 is unnecessary - it defaults to 0, 0. I may've missed QuestsCompletedInnerPanel setting its own Left, but I checked and it looked like it didn't. You repeat this a little below with the skill tree panel.

Reply:
I'm not sure how you were seeing this, as this does not exist anymore. Were you viewing a commit?

Reply:
Oh, yeah. I look through each commit, usually checking future ones if relevant. I guess I shouldn't lol